index.js

```diff
@@ -1,15 +1,23 @@
-var postcss = require('postcss');
+var postcss = require('postcss'),
+    parser = require('postcss-value-parser');
 
 module.exports = postcss.plugin('postcss-round-subpixels', function (opts) {
-
-
     return function (css) {
       css.walkDecls(function(decl){
-        //  any number of Numbers, a decimal, then 'px'
-        if( decl.value.match(/\d*?\.\d*?px/g) )
-          // those maths make it into a rounded Number,
-          // then we make it a string again by adding 'px'
-          decl.value = Math.round(parseFloat(decl.value)) + "px";
+        //  use the parser to walk through each individual value in the decl
+        decl.value = parser(decl.value).walk( function( node ){
+          //  we can totally skip things that aren't a pixel or "word" value
+          if( node.type === 'space' || node.type === 'function' || node.type === 'div' ){
+            return false;
+          }
+          //  any number of Numbers, a decimal, then 'px'
+          if(node.value.match(/\d*?\.\d*?px/g) ){
```

Comment:
Any reason not to use https://github.com/TrySound/postcss-value-parser#valueparserunitvalue?